virtcontainers/qemu_arm64.go

```diff
@@ -38,15 +41,41 @@ var kernelRootParams = []Param{
 	{"root", "/dev/vda1"},
 }
 
+// variables rather than consts to allow tests to modify them
+var interruptFile = "/proc/interrupts"
+//on aarch64, we support different gic interrupt controllers
+//maximum number of vCPUs depends on the GIC version, or on how
+//many redistributors we can fit into the memory map.
+var gicList = map[string]uint32{
+	"GICv2": uint32(8),
+	"GICv3": uint32(123),
+	"GICv4": uint32(123),
+}
+
 var supportedQemuMachines = []govmmQemu.Machine{
 	{
 		Type:    QemuVirt,
 		Options: defaultQemuMachineOptions,
 	},
 }
 
+// Logger returns a logrus logger appropriate for logging qemu-aarch64 messages
+func qemuArmLogger() *logrus.Entry {
+	return virtLog.WithField("subsystem", "qemu-aarch64")
+}
+
 // MaxQemuVCPUs returns the maximum number of vCPUs supported
 func MaxQemuVCPUs() uint32 {
+	bytes, err := ioutil.ReadFile(interruptFile)
+	if err != nil {
+		qemuArmLogger().WithError(err).Error("Failed to read /proc/interrrupts")
+	}
+	for gicType, vCPUs := range gicList {
+		pattern := regexp.MustCompile(`\b` + gicType + `\b`)
```

Comment:
The regex seems a bit too much here, maybe something like:
```go
if strings.Contains(string(bytes), gicType) {
    return vCPUs
}
```
might be simpler and cost less to process.

Reply:
when I added support for arm64 I was unable to start a VM with `maxcpus` > `actual number of physical CPUs`. I think it's a limitation of KVM, having said that, I believe that this PR will fail on systems with gicV3 and `actual number of physical CPUs` < 123. @Pennyzct let me know when this PR is ready, I'd like to take a look